packages/bun-usockets/src/crypto/openssl.c

```diff
@@ -299,32 +322,16 @@ void us_internal_ssl_handshake(struct us_internal_ssl_socket_t *s) {
   }
 }
 
-struct us_internal_ssl_socket_t *
-us_internal_ssl_socket_close(struct us_internal_ssl_socket_t *s, int code,
-                             void *reason) {
-  struct us_internal_ssl_socket_context_t *context =
-      (struct us_internal_ssl_socket_context_t *)us_socket_context(0, &s->s);
-
-  if (s->pending_handshake) {
-    s->pending_handshake = 0;
-    if (context->on_handshake != NULL) {
-      struct us_bun_verify_error_t verify_error = us_internal_verify_error(s);
-      context->on_handshake(s, 0, verify_error, context->handshake_data);
-    }
-  }
-
-  return (struct us_internal_ssl_socket_t *)us_socket_close(
-      0, (struct us_socket_t *)s, code, reason);
-}
-
 struct us_internal_ssl_socket_t *
 ssl_on_close(struct us_internal_ssl_socket_t *s, int code, void *reason) {
   struct us_internal_ssl_socket_context_t *context =
       (struct us_internal_ssl_socket_context_t *)us_socket_context(0, &s->s);
-  if (s->pending_handshake) {
-    s->pending_handshake = 0;
+
+  s->pending_handshake = 0;
+  if (s->ssl) {
```

Comment:
Why would we get in a state where ssl is a nullptr? And have you checked that SSL_free doesn’t already do this check? BoringSSL tends to be very careful with things like this so it would be unsurprising if they check this already

Reply:
will do a cleanup